### PR TITLE
Add ansible linting to Jenkins

### LINKED ISSUE #827: Add ansible linting to Jenkins
Much of another PR I'm working on keeps touching bits that do not match expected formatting which causes me to make changes that are not specific to the feature I'm adding. I fear that this will make my PR less comprehensible and harder to review, thus I'd like to correct all linting errors and add ansible linting to Jenkins.

To that end, I'm splitting off all the syntactic changes into a PR (#826) that is specific to this need. This will also add lint checking to Jenkins to help prevent this issue in the future.

### PR DESCRIPTION
**Pull request by joejulian**
* Fix ansible lint errors
* Add ansible lint checking to Jenkins

Fixes #827

### CODE REVIEW

**Comment by coffeepac** [comment]
@joejulian I believe you need to rebase this onto master to get etcd stability fixes and e2e notifications messages.

File: Jenkinsfile [hunk 1]
```diff
@@ -57,7 +59,9 @@ podTemplate(label: 'k2', containers: [
 
             // Unit tests
             stage('Test: Unit') {
-                kubesh 'true' // Add unit test call here
+                customContainer('ansible-lint') {
+                    kubesh 'ansible-lint ansible/*.yaml'
```
Comment (coffeepac) [inline Jenkinsfile]: is this recursive?

File: ansible/roles/kraken.access/tasks/generate-admin-cert.yaml [hunk 2]
```diff
@@ -10,26 +10,27 @@
...
   when: (cluster.kubeAuth.authz.rbac is defined)
 
-- name: Generate end-user admin csr 
+- name: Generate end-user admin csr
   command: >
     openssl req -new -key {{ config_base | expanduser }}/{{ cluster.name }}/certs/end-user.key
       -subj {{ endsubj }} -out {{ config_base | expanduser }}/{{ cluster.name }}/certs/end-user.csr
-        creates={{ config_base | expanduser }}/{{ cluster.name }}/certs/end-user.csr
+  args:
+    creates: "{{ config_base | expanduser }}/{{ cluster.name }}/certs/end-user.csr"
```
Comment (coffeepac) [inline ansible/roles/kraken.access/tasks/generate-admin-cert.yaml]: this isn't the only arg here.  it was the only arg without a leading dash and with an equal sign but this is inconsistent within this command.

File: ansible/roles/kraken.access/tasks/generate-admin-kubeconfig.yaml [hunk 3]
```diff
@@ -22,6 +22,12 @@
       --certificate-authority={{ config_base | expanduser }}/{{ cluster.name }}/certs/ca.pem
       --embed-certs=true
       --kubeconfig={{ kubeconfig }}
+  vars:
+    kconfig: "{{ kubeconfig | is_file | ternary(lookup('file',kubeconfig),'---') | from_yaml }}"
```
Comment (coffeepac) [inline ansible/roles/kraken.access/tasks/generate-admin-kubeconfig.yaml]: do you want to use this var in the command? does that work?

Comment (joejulian) [reply ansible/roles/kraken.access/tasks/generate-admin-kubeconfig.yaml]: Nope, this does not work.

**Comment by joejulian** [comment]
Regarding args, I agree it should be consistent. It's already inconsistent throughout the entire project. Fixing that and adding that custom lint rule is outside the scope of this change but I'll create an issue.

File: Jenkinsfile (hunk 1 above)
Comment (joejulian) [reply Jenkinsfile]: Yes, specifying *.yaml will recurse down through the entire playbooks.

**Comment by joejulian** [comment]
I would prefer this be a "rebase and merge" to retain the individual commits. I think they're useful history.

**Pull request merged**